### ISSUE TITLE
Implement First-Party Mode Setup Success Subtle Notification Component

### ISSUE THREAD

**Issue by hussain-t** [google/site-kit-wp#9648]
## Feature Description

Create a new `<FirstPartyModeSetupSuccessSubtleNotification />` component to display a success message upon successfully enabling First-party Mode. This subtle notification should confirm activation, remain visible until dismissed, and be implemented using the new notifications API.

For more details, refer to the [Design Doc](https://docs.google.com/document/d/1PHptJ1Js_1OZrk5kRwoB3zPB2KQrXLJV7-3izi-bB30/edit?tab=t.0#heading=h.it8csv2b4o4p) and the [Figma design](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj/First-party-mode?node-id=56-3071&m=dev).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A First-Party Mode Setup Success Subtle Notification should confirm that First-party mode has been successfully enabled, providing users with a dismissible acknowledgment.
* Notification Display Conditions:
    * The notification should appear **only once** after First-party mode is successfully enabled via the setup banner on the dashboard.
	* The notification should not persist across page reloads, regardless of whether it has been dismissed.
    * The notification should not display if it has been dismissed.
* Notification Content:
    * Title: "You successfully enabled First-party mode!"
    * Description: "You can always disable it in Analytics or Ads settings"
* Notification Actions:
    * A dismiss button should allow users to hide the notification permanently.
    * Button text: "Got it"
    * Once dismissed, the notification should not reappear.
* The notification should be implemented using the new notifications API.
* Styling and content should align with the specifications outlined in the [Figma design](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj/First-party-mode?node-id=56-3071&m=dev).

## Implementation Brief

- [ ] Register the notification to the [DEFAULT_NOTIFICATIONS](https://github.com/google/site-kit-wp/blob/440dcd6e8289a8776acc60ecb3c13f99bfd0ac5a/assets/js/googlesitekit/notifications/register-defaults.js#L44) in `assets/js/googlesitekit/notifications/register-defaults.js`.
  - Pass the slug as `setup-success-notification-fpm` along with following object.
  - `Component` should be `FirstPartyModeSetupSuccessSubtleNotification` as explained below.
  - Priority should be 10.
  - areaSlug should be `NOTIFICATION_AREAS.BANNERS_BELOW_NAV`.
  - viewContexts should be `VIEW_CONTEXT_MAIN_DASHBOARD`.
  - isDismissible should be set to `false`.
  - In checkRequirements method, do the following.
    - Get the value of `FPM_SHOW_SETUP_SUCCESS_NOTIFICATION` from `core/ui` store. Cast this value into boolean and return it from the `checkRequirements` function.

- [ ] In `onCTAClick` method implemented in #9647.
  - set the `FPM_SHOW_SETUP_SUCCESS_NOTIFICATION` to `true` in `core/ui` store before calling `dismissNotification` method. This will be used in the `FirstPartyModeSetupSuccessSubtleNotification` component to display the notification, it is useful because we only want to display the notification only once, regardless of whether it has been dismissed.
  - invalidate the resolution for `getQueuedNotifications()` like [this one](https://github.com/google/site-kit-wp/blob/440dcd6e8289a8776acc60ecb3c13f99bfd0ac5a/assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget/index.js#L114-L117) once the notification is dismissed.

- [ ] Create a new component `FirstPartyModeSetupSuccessSubtleNotification` inside `assets/js/components/notifications`.
  - This component will receive two properties, `id` which is the slug passed in above step `setup-success-notification-fpm` and `Notification` component.
  - Wrap the `SubtleNotification` within `Notification` component and pass following props.
    - Pass the `title` and `description` props as per the figma designs to the `SubtleNotification` component.
    - `dismissCTA` should accept the `Dismiss` component with following props.
      - `id` as the one which we received as parameter.
      - `primary` set to `false`.
      - `dismissLabel` set to `Got it`.

- [ ] Make sure that styling and typography matches the Figma design.

### Test Coverage

1. Create a story for the `FirstPartyModeSetupSuccessSubtleNotification` component.
2. Write JS tests for `FirstPartyModeSetupSuccessSubtleNotification` component including various scenarios for displaying not displaying the notifications.

## QA Brief

* Follow steps 1, 2, 3 and 5 from the QAB for https://github.com/google/site-kit-wp/issues/9647 in order to show the FPM setup banner and enable FPM via its primary CTA.
* Verify the success notification appears as per the AC.
* Please note, the styling should be consistent with other [subtle notifications](https://google.github.io/site-kit-wp/storybook/develop/?path=/story/components-subtlenotification--success-with-description) - at the time of writing the vertical padding and gap between the icon and text is inconsistent in the Figma design, although I've [asked Sigal to update it](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj?node-id=56-2795&m=dev#1056620654).

## Changelog entry

* Add first-party mode setup success notification.

**Comment by techanvil** [comment]
Hey @hussain-t, I think we should tweak the AC to specify that the banner should only appear when first-party mode has been enabled via the setup banner on the dashboard, and should not appear on the next page load regardless of whether the user has dismissed it.

We don't need to persistently show the banner when the page reloads (we _are_ doing this for AS but this was an oversight and we intend to make the AS banner follow the above behaviour in a future update).

**Comment by hussain-t** [comment]
Thanks, @techanvil! I've updated the AC accordingly.

**Comment by techanvil** [comment]
Thanks @hussain-t! That LGTM.

AC ✅

**Comment by ankitrox** [comment]
@hussain-t @techanvil I've added #9647 as the dependency for this issue based on the drafted IB. Please let me know if this is fine.

Thank you.

**Comment by techanvil** [comment]
Thanks @ankitrox, that's a good shout 👍

**Comment by ankitrox** [comment]
Thanks @techanvil for reviewing the IB and providing valuable insights. 

I have updated the IB as per feedback and also created https://github.com/google/site-kit-wp/issues/9740 to use on-demand notification.

**Comment by wpdarren** [comment]
### QA Update: ⚠ 

@techanvil I have a UX/UI observation. 

As you can see from the screencast, when I click on the `Enable First-party mode ` CTA button, there's no spinning loading icon, like our other banner notifications.  I realise that is probably out of scope for this ticket, but after a few seconds the banner disappears and I have to scroll up the page to see the subtle success notification.   Interested in your thoughts on this one. 

https://github.com/user-attachments/assets/0272ad03-4ec5-4ddd-9578-a7ba2517feda

**Verified:**

- First-Party Mode Setup Success Subtle Notification appears to confirm that First-party mode has been successfully enabled, providing users with a dismissible acknowledgment:
  - Notification Display Conditions:
   - The notification appears only once after First-party mode is successfully enabled via the setup banner on the dashboard.
   - The notification does not persist across page reloads, regardless of whether it has been dismissed.
   - The notification does not display if it has been dismissed.
  - Notification Content:
   - Title: `You successfully enabled First-party mode!`
   - Description: `You can always disable it in Analytics or Ads settings`
  - Notification Actions:
   - A dismiss button allows users to hide the notification permanently.
   - Button text: `Got it`
   - Once dismissed, the notification should not reappear.
- Styling and content aligns with the specifications outlined in the [Figma design](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj/First-party-mode?node-id=56-3071&m=dev).

![Image](https://github.com/user-attachments/assets/75c6d89e-d67c-4767-96a5-2397b062d035)

**Comment by techanvil** [comment]
Hi @wpdarren, thanks for flagging these issues.

The point about there being no loading state is a good one, however as you've surmised it is out of scope here. Please can you raise a new issue for it?

As for having to scroll up to see the success notification, this is also a good point, however it looks like this is occurring in your example due to the Consent Mode banner being present on the page - when I try without the CoMo banner present the screen remains at the top with the success notification immediately visible.

https://github.com/user-attachments/assets/3dbadcbe-ce15-4d2b-9229-5b7e8eb696e3

This should be resolved when the Consent Mode and other relevant banners have been migrated to the new notifications infrastructure, ensuring that only one banner is present on the page at a time.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

@techanvil thank you!  I will create a ticket for the CTA on the banner notification. 

**Verified:**

- First-Party Mode Setup Success Subtle Notification appears to confirm that First-party mode has been successfully enabled, providing users with a dismissible acknowledgment:
  - Notification Display Conditions:
   - The notification appears only once after First-party mode is successfully enabled via the setup banner on the dashboard.
   - The notification does not persist across page reloads, regardless of whether it has been dismissed.
   - The notification does not display if it has been dismissed.
  - Notification Content:
   - Title: `You successfully enabled First-party mode!`
   - Description: `You can always disable it in Analytics or Ads settings`
  - Notification Actions:
   - A dismiss button allows users to hide the notification permanently.
   - Button text: `Got it`
   - Once dismissed, the notification should not reappear.
- Styling and content aligns with the specifications outlined in the [Figma design](https://www.figma.com/design/udlPW35VZ0EDnSbRxYOGMj/First-party-mode?node-id=56-3071&m=dev).

![Image](https://github.com/user-attachments/assets/75c6d89e-d67c-4767-96a5-2397b062d035)
